core/domain/exp_services.py

```diff
@@ -1723,6 +1723,496 @@ def is_voiceover_change_list(change_list):
     return True
 
 
+def get_composite_change_list(exp_id, frontend_version, backend_version):
+    """Returns a list of ExplorationChange domain objects consisting of
+    changes from frontend_version to backend_version in an exploration.
+
+    Args:
+        exp_id: str. The id of the exploration.
+        frontend_version: int. The version of the exploration on which the user
+            is working.
+        backend_version: int. The latest version of the exploration from the
+            backend.
+
+    Returns:
+        list(ExplorationChange). List of ExplorationChange domain objects
+        consisting of changes from frontend_version to backend_version.
+    """
+    if frontend_version > backend_version:
+        raise Exception(
+            'Unexpected error: Trying to find change list from version %s '
+            'of exploration to version %s. Please reload and try again.'
+            % (frontend_version, backend_version))
+
+    snapshots_metadata = get_exploration_snapshots_metadata(exp_id)
+    composite_change_list_dict = []
+    for i in python_utils.RANGE(frontend_version, backend_version):
+        composite_change_list_dict += snapshots_metadata[i]['commit_cmds']
```

Comment:
Just to confirm, snapshots_metadata are returned, sorted by version, right?
Even so, I think it might be better to use the ID of the snapshots instead of depending on the sorting of the list. I think snapshot IDs are in the form `<exp_id>-<exp_version>`, which might be easier, and if anything is missing there, throw an exception or log the error. @atpug22 @kevintab95 What do you think?

Reply:
Here this get_exploration_snapshots_metadata returns the list according to the version number only in an increasing order one by one. They pass the version numbers to the get_snapshots_metadata function in exp_models and there they check for each version and then send otherwise raise the error. So the list which we are receiving is already checked and sorted on the basis of version number. So I don't think so we really need to check it again.